javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -490,15 +491,23 @@ default T get() {
      *
      * @return {@code Some(traversable)} or {@code None} if this is empty.
      */
-    Option<? extends Traversable<T>> initOption();
+    default Option<? extends Traversable<T>> initOption() {
+        return whenNotEmpty(this::init);
+    }
+
+    default <R> Option<R> whenNotEmpty(Supplier<R> supplier) {
```

Comment:
if we don't want to expose this (i.e. no `protected` in interfaces), maybe we could import it statically (though we won't be able to access `isEmpty` then, and there would be no advantage of having this method...